core/src/Cache/TokenCacheSerializeHelper.cs

```diff
@@ -0,0 +1,132 @@
+        /// </summary>
+        /// <param name="tokenCacheAccessor">Token cache accessor to perform cache write operations (to fill-in from the state)</param>
+        /// <param name="state">Array of bytes containing serialized cache data</param>
+        /// <remarks>
+        /// <paramref name="state"/>Is a Json blob containing access tokens, refresh tokens, id tokens and accounts information
+        /// </remarks>
+        public static void DeserializeMsalCache(TokenCacheAccessor tokenCacheAccessor, byte[] state, RequestContext requestContext)
+        {
+            Dictionary<string, IEnumerable<string>> cacheDict = JsonHelper
+                .DeserializeFromJson<Dictionary<string, IEnumerable<string>>>(state);
+
+            if (cacheDict == null || cacheDict.Count == 0)
+            {
+                string msg = "Msal Cache is empty.";
+                CoreLoggerBase.Default.Info(msg);
+                CoreLoggerBase.Default.InfoPii(msg);
+                return;
+            }
+
+            if (cacheDict.ContainsKey(AccessTokenKey))
+            {
+                foreach (var atItem in cacheDict[AccessTokenKey])
+                {
+                    var msalAccessTokenCacheItem = JsonHelper.TryToDeserializeFromJson<MsalAccessTokenCacheItem>(atItem, requestContext);
+                    if (msalAccessTokenCacheItem != null)
+                    {
+                        tokenCacheAccessor.SaveAccessToken(msalAccessTokenCacheItem);
+                    }
+                }
+            }
+
+            if (cacheDict.ContainsKey(RefreshTokenKey))
+            {
+                foreach (var rtItem in cacheDict[RefreshTokenKey])
+                {
+                    var msalRefreshTokenCacheItem = JsonHelper.TryToDeserializeFromJson<MsalRefreshTokenCacheItem>(rtItem, requestContext);
+                    if (msalRefreshTokenCacheItem != null)
+                    {
+                        tokenCacheAccessor.SaveRefreshToken(msalRefreshTokenCacheItem);
+                    }
+                }
+            }
+
+            if (cacheDict.ContainsKey(IdTokenKey))
+            {
+                foreach (var idItem in cacheDict[IdTokenKey])
+                {
+                    var msalIdTokenCacheItem = JsonHelper.TryToDeserializeFromJson<MsalIdTokenCacheItem>(idItem, requestContext);
+                    if (msalIdTokenCacheItem != null)
+                    {
+                        tokenCacheAccessor.SaveIdToken(msalIdTokenCacheItem);
+                    }
+                }
+            }
+
+            if (cacheDict.ContainsKey(AccountKey))
+            {
+                foreach (var account in cacheDict[AccountKey])
+                {
+                    var msalAccountCacheItem = JsonHelper.TryToDeserializeFromJson<MsalIdTokenCacheItem>(account, requestContext);
+
+                    tokenCacheAccessor.SaveAccount(JsonHelper.DeserializeFromJson<MsalAccountCacheItem>(account));
```

Comment:
`msalAccountCacheItem` is not being used...something doesn't look right here.

Reply:
good catch  🐠